src/wallet/wallet.cpp

```diff
@@ -3323,18 +3323,22 @@ bool CWallet::CreateCollateralTransaction(CMutableTransaction& txCollateral, std
         return false;
     }
 
-    // make our change address
-    CScript scriptChange;
-    CPubKey vchPubKey;
-    assert(reservekey.GetReservedKey(vchPubKey, true)); // should never fail, as we just unlocked
-    scriptChange = GetScriptForDestination(vchPubKey.GetID());
-    reservekey.KeepKey();
-
     txCollateral.vin.push_back(txdsinCollateral);
 
     //pay collateral charge in fees
-    CTxOut txout = CTxOut(nValue - CPrivateSend::GetCollateralAmount(), scriptChange);
-    txCollateral.vout.push_back(txout);
+    if (nValue >= CPrivateSend::GetCollateralAmount()*2) {
+        // make our change address
+        CScript scriptChange;
+        CPubKey vchPubKey;
+        assert(reservekey.GetReservedKey(vchPubKey, true)); // should never fail, as we just unlocked
```

Comment:
Not introduced by this PR, but in Bitcoin they removed all uses of `assert(doImportantLogic())` and replaced it with `if(!doImportantLogic()) assert(false);` or `bool success = doImportantLogic(); assert(success);`
I assume they started doing this to allow release builds which omit all asserts.